app/validators/routing/types.py

```diff
@@ -64,5 +64,19 @@ def resolve_value_source_json_type(value_source, schema):
     return TYPE_STRING
 
 
+def resolve_value_source_calculation_type(value_source, schema):
```

Comment:
Where is this actually used apart from tests...?

Reply:
Only used for tests at the moment. Written especially for tests.

Reply:
Ok, removed, only used in code now, only in test.

Reply:
But it isn't actually used in code thought?. And the tests are redundant right now cause it is just checking the type of the answer id...?.

We need to check if an answer source is used, its answer type resolves to a number.

Reply:
Don't we already do that? In this test...
```python
def test_resolve_answer_value_source_calculation_type():
    filename = "schemas/valid/test_valid_value_source_calculations.json"

    questionnaire_schema = QuestionnaireSchema(_open_and_load_schema_file(filename))

    value_source = {"source": "answers", "identifier": "total-answer"}

    answer_id = value_source["identifier"]
    answer_type = (
        questionnaire_schema.answers_with_context[answer_id]["answer"]["type"]
    ).lower()

    assert answer_type == TYPE_NUMBER
```
...we check if answer source is used and if yes the answer type is checked if it's number?
Same for calculated summary source.

Reply:
Those tests are manually checking the answer source type is `TYPE_NUMBER` it is not validating that if an answer source is used, it is a number. That passes purely cause your valid schema is using number type answer source. My point is that,  I don't see any checks in code to ensure we check the types. If you run an invalid schema where the source was textfield, nothing would break....?

Reply:
So, I've added the validation now, checks sources answers/summaries resolved types. `CalculatedQuestionValidator` needed some overhaul and the problem was to get the wider context of the schema. Fixed the tests to cover the changes. Ready for re-review.